pymeasure/instruments/common_base.py

```diff
@@ -0,0 +1,569 @@
+
+    This class contains everything needed for pymeasure's property creator
+    :meth:`control` and its derivatives :meth:`measurement` and :meth:`setting`.
+    """
+
+    # Variable holding the list of DynamicProperty parameters that are configurable
+    # by users
+    _fget_params_list = ('get_command',
+                         'values',
+                         'map_values',
+                         'get_process',
+                         'command_process',
+                         'check_get_errors')
+
+    _fset_params_list = ('set_command',
+                         'validator',
+                         'values',
+                         'map_values',
+                         'set_process',
+                         'command_process',
+                         'check_set_errors')
+
+    # Prefix used to store reserved variables
+    __reserved_prefix = "___"
+
+    def __init__(self):
+        self._special_names = self._setup_special_names()
+        # Add children from ChildDescriptors.
+        for item, value in inspect.getmembers(self.__class__):
+            if isinstance(value, ChildDescriptor):
+                if isinstance(value.id, (list, tuple)):
+                    for id, cls in zip(value.id, value.cls):
+                        self.add_child(cls, id, collection=item, **value.kwargs)
+                else:
+                    self.add_child(value.cls, value.id, collection=item, **value.kwargs)
+
+    @staticmethod
+    def children(id, cls, **kwargs):
+        """Describe children of this parent class.
+
+        The children will be added to the parent instance at instantiation with
+        :meth:`add_child`. The variable name (e.g. :code:`channels`) will be
+        used as the `collection` of the children. You may define the attribute
+        prefix via keyword arguments. Normally, do use :code:`channels` as variable
+        and do leave the prefix at the default :code:`"ch"`.
+
+        .. code::
+
+            class SomeInstrument(Instrument):
+                # Three channels of the same type: 'ch_A', 'ch_B', 'ch_C' in 'channels'
+                channels = CommonBase.children(["A", "B", "C"], ChildClass)
+                # Two functions of different types: 'fn_power', 'fn_voltage' in 'functions'
+                functions = CommonBase.children(["power", "voltage"],
+                                                (PowerChannel, VoltageChannel), prefix="fn")
+                # A control without a prefix: 'motor'
+                motor = CommonBase.children(None, MotorControl)
+
+        :param id: Single value or tuple/list of ids of the children. If `None`,
+            the single child will be added directly under the variable name.
+        :param cls: Class for all children or tuple/list of classes, one for each child.
+        :param \\**kwargs: Keyword arguments for all children.
+        """
+        try:
+            valid_class = issubclass(cls, CommonBase)
+        except Exception:
```

Comment:
```suggestion
        except TypeError:
```
I'm surprised that our linter did not flag this. My internal one did, at least. :D
Don't catch `Exception`, (nearly) always be specific what you want to catch.

Reply:
Done